tests/simplewaveequation.py

```diff
@@ -0,0 +1,236 @@
+    :param tmax: define simulation time
+    :param output_vts: switch for code to output at each time step
+    (such as save field as vtk file), default False (no output)
+    :param o_converge: switch for code to calculate L2 norms
+    for convergence test, default True (output)
+    :param omp: swtich for inserting #pragma omp for before outer loop
+    default True (use omp)
+    :param simd: switch for inserting #pragma simd before inner loop
+    default False (do not use simd)
+    :param ivdep: switch for inserting #praga ivdep before inner loop
+    default True (use ivdep)
+    default False (not include vtk header files)
+    :param double: switch for using double as real number variables
+    default False (use float)
+    :param expand: expand kernel fully (no factorisation), default True
+    :param eval_const: evaluate all constants in kernel in generated code default True
+    :param filename: output source code file name as string
+    :param pluto: switch for inserting #pragma scop and #pragma endscop for
+    pluto optimisation
+    :param fission: switch for doing loop fission optimisation
+    """
+
+    print 'domain size: ' + str(domain_size)
+    print 'grid size: ' + str(grid_size)
+    print 'approximation order: ' + str(accuracy_order)
+    print 'dt: ' + str(dt)
+    print 'tmax: ' + str(tmax)
+
+    # Declare fields
+    MAIN_GRID = RegularField('MAIN_GRID', dimension=3)
+
+    grid = RegularGrid(dimension=3, domain_size=domain_size, grid_size=grid_size, fields=[MAIN_GRID], pluto=pluto, fission=fission)
+    grid.set_time_step(dt, tmax)
+    grid.set_switches(omp=omp, simd=simd, ivdep=ivdep, double=double,
+                      expand=expand, eval_const=eval_const,
+                      output_vts=output_vts, converge=o_converge)
+    # define parameters
+    # const_c: The constant from the wave equation
+    x, y, z, const_c = symbols('x y z c')
+    grid.set_index([x, y, z])
+    # Set the values of the constant C from the wave equation, and the initial velocity (single scalar value for all dimensions=0)
+    grid.set_params(c=2, v=1)
+
+    # Check if its possible to calculate a limit for the simple case
+    # print 'require dt < ' + str(grid.get_time_step_limit())
+
+    # Initial wave displacements
+    # In the cube for t=0, the initial displacement values are accepted as a function of x, y and z.
+    # Here we use sin(x+y+z) to populate the initial displacements
+    MAIN_GRID_init_func = sin(x+y+z)
+    MAIN_GRID.set_analytic_solution(MAIN_GRID_init_func)
+    grid.set_order(accuracy_order)
+
+    # The equation involves second-order derivatives, so instruct the class not to drop second derivatives from the analysis
+    grid.calc_derivatives(2)
+
+    # The DE that generates the kernel
+    eq0 = Eq(MAIN_GRID.d[0][2], (const_c**2)*(MAIN_GRID.d[1][2] + MAIN_GRID.d[2][2] + MAIN_GRID.d[2][2]))
+
+    grid.solve_fd([eq0])
+    # print 'kernel Weighted AI: ' + '%.2f' % grid.get_overall_kernel_ai()[1]
```

Comment:
Another line that was present in eigenwave.py that I had my doubts about. The next line calculates the Kernel AI. Since there is only one kernel in this version, the weighted AI doesn't make sense any more in my opinion. The grid class method that this is calling doesn't exist for the regular grid. I am happy to remove it.